ipn/ipnlocal/local.go

```diff
@@ -4093,3 +4012,33 @@ func (b *LocalBackend) SetDevStateStore(key, value string) error {
 func (b *LocalBackend) ShouldInterceptTCPPort(port uint16) bool {
 	return b.shouldInterceptTCPPortAtomic.Load()(port)
 }
+
+func (b *LocalBackend) SwitchProfile(profile ipn.ProfileID) error {
+	if b.CurrentProfile().ID == profile {
+		return nil
+	}
+	b.mu.Lock()
+	if err := b.pm.SwitchProfile(profile); err != nil {
+		b.mu.Unlock()
+		return err
+	}
+	b.setNetMapLocked(nil)                 // Reset netmap.
+	b.enterStateLockedOnEntry(ipn.NoState) // Reset state.
+	return b.Start(ipn.Options{})
+}
+
+func (b *LocalBackend) DeleteProfile(p ipn.ProfileID) error {
+	return b.pm.DeleteProfile(p)
+}
+
+func (b *LocalBackend) CurrentProfile() ipn.LoginProfile {
+	return b.pm.CurrentProfile()
+}
+
+func (b *LocalBackend) NewProfile() {
+	b.pm.NewProfile()
+}
```

Comment:
this is mysterious without a return value or docs

Reply:
now with docs